comm.cpp

```diff
@@ -195,6 +204,12 @@ int Endpoint::read_msg(struct buffer *pbuf)
         if (rx_buf.len < sizeof(*hdr))
             return 0;
 
+        msg_id = hdr->msgid;
+        payload = rx_buf.data + sizeof(*hdr);
+
+        *sysid = hdr->sysid;
+        *compid = hdr->compid;
```

Comment:
I think this should be a member of Endpoint instead of reading it from the arguments.... And we only set it once and emit a warning (possibly throttled not to spam the console) if it's read again and different from what's already saved.